python/vineyard/cli.py

```diff
@@ -452,20 +465,53 @@ def debug(client, args):
     print(f'The result returned by the debug handler:\n{result}')
 
 
-def get_tree(meta, tree, parent=None):
+def get_tree(meta, tree, memory=False, memory_dict=None, parent=None):
     """Utility to display object lineage in a tree like form."""
     if parent is None:
-        parent = f'<{meta["typename"]}>:{meta["id"]}'
+        parent = f'{meta["typename"]} <{meta["id"]}>'
+        if memory:
+            memory_used = get_memory(meta["nbytes"])
+            parent += f': {memory_used}'
+            memory_dict[parent] = memory_used
         tree.create_node(parent, parent)
     else:
-        new_parent = f'<{meta["typename"]}>:{meta["id"]}'
+        new_parent = f'{meta["typename"]} <{meta["id"]}>'
+        if memory:
+            memory_used = get_memory(meta["nbytes"])
+            new_parent += f': {memory_used}'
+            memory_dict[new_parent] = memory_used
```

Comment:
These lines can also be reorganized to avoid duplicates, like
```
new_parent = xxx
if memory:
   ...
tree.create_node(new_parent, new_parent, parent=parent)
parent = new_parent
```
Since the `create_node(self, tag=None, identifier=None, parent=None)` have this default `parent=None` you can always create a new parent and pass the old parent to it.